Sources/JSON.swift

```diff
@@ -295,6 +295,42 @@ extension Dictionary {
     }
 }
 
+// MARK: - Raw
+
+extension JSON: RawRepresentable {
+    
+    enum JSONDataError: ErrorType {
```

Comment:
We'll need to make this error type public, to be used outside the framework. Also, let's go ahead and change the name of this type to `DataError`, since it's already namespaced to `JSON`. That way, instead of `JSON.JSONDataError`, you'd simply do `JSON.DataError`

Also, for consistency, let's name these cases `MissingObject` and `InvalidObject`

```swift
public enum DataError: ErrorType {
   case MissingObject
   case InvalidObject
}
```